File: gluecodium/src/test/resources/smoke/field_constructors/output/swift/smoke/FieldConstructorWithDeprecationAndComment.swift
//
//
import Foundation
public struct FieldConstructorWithDeprecationAndComment {
    public var stringField: String
    /// Some field constructor
    /// - Parameters
    ///   - stringField:
    @available(*, deprecated, message: "Shouldn't really use it")
    public init(stringField: String) {
        self.stringField = stringField
    }
    internal init(cHandle: _baseRef) {
        stringField = moveFromCType(smoke_FieldConstructorWithDeprecationAndComment_stringField_get(cHandle))
    }
}
internal func copyFromCType(_ handle: _baseRef) -> FieldConstructorWithDeprecationAndComment {
    return FieldConstructorWithDeprecationAndComment(cHandle: handle)
}
internal func moveFromCType(_ handle: _baseRef) -> FieldConstructorWithDeprecationAndComment {
    defer {
        smoke_FieldConstructorWithDeprecationAndComment_release_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: FieldConstructorWithDeprecationAndComment) -> RefHolder {
    let c_stringField = moveToCType(swiftType.stringField)
    return RefHolder(smoke_FieldConstructorWithDeprecationAndComment_create_handle(c_stringField.ref))
}
internal func moveToCType(_ swiftType: FieldConstructorWithDeprecationAndComment) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_FieldConstructorWithDeprecationAndComment_release_handle)
}
internal func copyFromCType(_ handle: _baseRef) -> FieldConstructorWithDeprecationAndComment? {
    guard handle != 0 else {
        return nil
    }
    let unwrappedHandle = smoke_FieldConstructorWithDeprecationAndComment_unwrap_optional_handle(handle)
    return FieldConstructorWithDeprecationAndComment(cHandle: unwrappedHandle) as FieldConstructorWithDeprecationAndComment
}
internal func moveFromCType(_ handle: _baseRef) -> FieldConstructorWithDeprecationAndComment? {
    defer {
        smoke_FieldConstructorWithDeprecationAndComment_release_optional_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: FieldConstructorWithDeprecationAndComment?) -> RefHolder {
    guard let swiftType = swiftType else {
        return RefHolder(0)
    }
    let c_stringField = moveToCType(swiftType.stringField)
    return RefHolder(smoke_FieldConstructorWithDeprecationAndComment_create_optional_handle(c_stringField.ref))
}
internal func moveToCType(_ swiftType: FieldConstructorWithDeprecationAndComment?) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_FieldConstructorWithDeprecationAndComment_release_optional_handle)
}

File: gluecodium/src/test/resources/smoke/field_constructors/output/dart/lib/src/smoke/field_constructor_with_deprecation_and_comment.dart
import 'dart:ffi';
import 'package:library/src/_library_context.dart' as __lib;
import 'package:library/src/builtin_types__conversion.dart';
class FieldConstructorWithDeprecationAndComment {
  String stringField;
  /// Some field constructor
  /// [stringField]
  @Deprecated("Shouldn't really use it")
  FieldConstructorWithDeprecationAndComment(this.stringField);
}
// FieldConstructorWithDeprecationAndComment "private" section, not exported.
final _smokeFieldconstructorwithdeprecationandcommentCreateHandle = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Pointer<Void> Function(Pointer<Void>),
    Pointer<Void> Function(Pointer<Void>)
  >('library_smoke_FieldConstructorWithDeprecationAndComment_create_handle'));
final _smokeFieldconstructorwithdeprecationandcommentReleaseHandle = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Void Function(Pointer<Void>),
    void Function(Pointer<Void>)
  >('library_smoke_FieldConstructorWithDeprecationAndComment_release_handle'));
final _smokeFieldconstructorwithdeprecationandcommentGetFieldstringField = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Pointer<Void> Function(Pointer<Void>),
    Pointer<Void> Function(Pointer<Void>)
  >('library_smoke_FieldConstructorWithDeprecationAndComment_get_field_stringField'));
Pointer<Void> smokeFieldconstructorwithdeprecationandcommentToFfi(FieldConstructorWithDeprecationAndComment value) {
  final _stringFieldHandle = stringToFfi(value.stringField);
  final _result = _smokeFieldconstructorwithdeprecationandcommentCreateHandle(_stringFieldHandle);
  stringReleaseFfiHandle(_stringFieldHandle);
  return _result;
}
FieldConstructorWithDeprecationAndComment smokeFieldconstructorwithdeprecationandcommentFromFfi(Pointer<Void> handle) {
  final _stringFieldHandle = _smokeFieldconstructorwithdeprecationandcommentGetFieldstringField(handle);
  try {
    return FieldConstructorWithDeprecationAndComment(
      stringFromFfi(_stringFieldHandle)
    );
  } finally {
    stringReleaseFfiHandle(_stringFieldHandle);
  }
}
void smokeFieldconstructorwithdeprecationandcommentReleaseFfiHandle(Pointer<Void> handle) => _smokeFieldconstructorwithdeprecationandcommentReleaseHandle(handle);
// Nullable FieldConstructorWithDeprecationAndComment
final _smokeFieldconstructorwithdeprecationandcommentCreateHandleNullable = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Pointer<Void> Function(Pointer<Void>),
    Pointer<Void> Function(Pointer<Void>)
  >('library_smoke_FieldConstructorWithDeprecationAndComment_create_handle_nullable'));
final _smokeFieldconstructorwithdeprecationandcommentReleaseHandleNullable = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Void Function(Pointer<Void>),
    void Function(Pointer<Void>)
  >('library_smoke_FieldConstructorWithDeprecationAndComment_release_handle_nullable'));
final _smokeFieldconstructorwithdeprecationandcommentGetValueNullable = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Pointer<Void> Function(Pointer<Void>),
    Pointer<Void> Function(Pointer<Void>)
  >('library_smoke_FieldConstructorWithDeprecationAndComment_get_value_nullable'));
Pointer<Void> smokeFieldconstructorwithdeprecationandcommentToFfiNullable(FieldConstructorWithDeprecationAndComment? value) {
  if (value == null) return Pointer<Void>.fromAddress(0);
  final _handle = smokeFieldconstructorwithdeprecationandcommentToFfi(value);
  final result = _smokeFieldconstructorwithdeprecationandcommentCreateHandleNullable(_handle);
  smokeFieldconstructorwithdeprecationandcommentReleaseFfiHandle(_handle);
  return result;
}
FieldConstructorWithDeprecationAndComment? smokeFieldconstructorwithdeprecationandcommentFromFfiNullable(Pointer<Void> handle) {
  if (handle.address == 0) return null;
  final _handle = _smokeFieldconstructorwithdeprecationandcommentGetValueNullable(handle);
  final result = smokeFieldconstructorwithdeprecationandcommentFromFfi(_handle);
  smokeFieldconstructorwithdeprecationandcommentReleaseFfiHandle(_handle);
  return result;
}
void smokeFieldconstructorwithdeprecationandcommentReleaseFfiHandleNullable(Pointer<Void> handle) =>
  _smokeFieldconstructorwithdeprecationandcommentReleaseHandleNullable(handle);
// End of FieldConstructorWithDeprecationAndComment "private" section.
